src/System.Text.Json/src/System/Text/Json/Node/JsonString.cs

```diff
@@ -208,5 +209,55 @@ public string Value
         ///   Returns <see cref="JsonValueKind.String"/>
         /// </summary>
         public override JsonValueKind ValueKind { get => JsonValueKind.String; }
+
+        /// <summary>
+        ///   Converts the text value of this instance, which should encode binary data as base-64 digits, to an equivalent 8-bit unsigned <see cref="byte"/> array.
+        ///   The return value indicates wether the conversion succeeded.
+        /// </summary>
+        /// <param name="value">
+        ///   When this method returns, contains the <see cref="byte"/> array equivalent of the text contained in this instance,
+        ///   if the conversion succeeded.
+        /// </param>
+        /// <returns>
+        ///   <see langword="true"/> if text was converted successfully; othwerwise returns <see langword="false"/>.
+        /// </returns>
+        internal bool TryGetBytesFromBase64(out byte[] value)
+        {
+            if (_value.Length < 1)
+            {
+                value = default;
+                return false;
+            }
+
+            // we decode string -> byte, so the resulting length will
+            // be /4 * 3 - padding. To be on the safe side, keep padding in slice later
+            int bufferSize = _value.Length / 4 * 3;
```

Comment:
Another approach that might be worth exploring is transcoding the underlying string to UTF-8 (rather than UTF-16) and use the built-in Base64 UTF-8 APIs.

Those APIs also provide the buffer size needed (and are available on netstandard2.0/generally faster). Might lead to simpler implementation (though I don't know if it will be too cost-prohibitive in terms of perf to transcode up-front).

`GetMaxDecodedFromUtf8Length`
https://github.com/dotnet/corefx/blob/38aa9151607cb63bb569beb42880057ba2cd3da0/src/System.Memory/src/System/Buffers/Text/Base64Decoder.cs#L230

https://github.com/dotnet/corefx/blob/38aa9151607cb63bb569beb42880057ba2cd3da0/src/System.Memory/src/System/Buffers/Text/Base64Decoder.cs#L253

https://apisof.net/catalog/System.Buffers.Text.Base64.GetMaxDecodedFromUtf8Length(Int32)
https://apisof.net/catalog/System.Buffers.Text.Base64.DecodeFromUtf8InPlace(Span%3CByte%3E,Int32)

Reply:
I was thinking something like this, but that isn't great for performance for smaller strings (< 100 characters), but is on-par for large strings (> 1000 characters).

**So, I would keep the current approach you have.**

```C#
internal bool TryGetBytesFromBase64New(string _value, out byte[] value)
{
    if (_value.Length < 4)
    {
        value = default;
        return false;
    }

    // There might be some concerns about what the maximum value of _value can be and if this can integer overflow
    // This would need to be validated first before going down this approach.
    int bufferSize = _value.Length * 3; // max expansion due to transcoding, use JsonConstants.MaxExpansionFactorWhileTranscoding
    bufferSize = Base64.GetMaxDecodedFromUtf8Length(bufferSize);

    byte[] arrayToReturnToPool = null;
    try
    {
        Span<byte> buffer = bufferSize <= 256
            ? stackalloc byte[bufferSize]
            : arrayToReturnToPool = ArrayPool<byte>.Shared.Rent(bufferSize);

        OperationStatus status = Utf8.FromUtf16(_value.AsSpan(), buffer, out int charsRead, out int bytesWritten);
        // Can't have invalid UTF-8 data and since we define the destination size as max, can't have needs more data/destination too small
        Debug.Assert(status == OperationStatus.Done);
        Debug.Assert(charsRead == _value.Length);

        status = Base64.DecodeFromUtf8InPlace(buffer.Slice(0, bytesWritten), out bytesWritten);
        if (status == OperationStatus.Done)
        {
            value = buffer.Slice(0, bytesWritten).ToArray();
            return true;
        }
        else
        {
            Debug.Assert(status == OperationStatus.InvalidData);
            throw new FormatException("Invalid Base64 - use better message");
        }
    }
    finally
    {
        if (arrayToReturnToPool != null)
        {
            ArrayPool<byte>.Shared.Return(arrayToReturnToPool);
        }
    }
}
```